### ISSUE TITLE
aws-logs: Allow JSON filter patterns to use regex by removing automatic double quotes around pattern strings

### ISSUE THREAD

**Issue by sobesim** [aws/aws-cdk#30451]
### Describe the feature

[FilterPattern.stringValue](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-logs.FilterPattern.html) should allow the user to specify if the `value` input field is a regex string or not.

### Use Case

According to [CloudWatch filter pattern documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html#regex-expressions), regex expressions must be surrounded by `%`. 

I'm trying to use cdk's [FilterPattern.stringValue](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-logs.FilterPattern.html) to match a json document. 

CDK automatically adds double quotes in the final filter pattern, which prevents Cloudwatch from recognizing any string as a regex pattern. For example:
- Correct: { $.foo = %bar% }
- Incorrect: { $.foo = “%bar%” }, what is produced by CDK and does not match the document : {“foo”:“bar”}

### Proposed Solution

Reason for this behavior may be this line: https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-logs/lib/pattern.ts#L399. Perhaps we need another input to `stringValue` that allows the user to specify if the input string is a regex pattern or not.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.144.0

### Environment details (OS name and version, etc.)

Linux

**Comment by brian-stack** [comment]
Also impacted by this issue.

**Comment by jan-xyz** [comment]
I created a PR and maybe one of you is willing to also provide some feedback: https://github.com/aws/aws-cdk/pull/30741

**Comment by studds** [comment]
Is there any workaround for this?

**Comment by jan-xyz** [comment]
Yes, provide your own function that adds a regex filter pattern, like the one from the PR I linked.

**Comment by jan-xyz** [comment]
@pahud anything I can do to reopen the PR I made? It was far from abandoned. I requested support weeks ago and never got any response. Honestly, this felt super discouraging and I don't think I will contribute to CDK in the future.

**Comment by paulhcsun** [comment]
Hi @jan-xyz, apologies for the late response on your PR. I've replied on the PR with the information you've requested. Thank you for the contribution!

**Comment by JevinAnderson** [comment]
Hi @jan-xyz , thanks for raising this! Found this while trying to resolve my own similar problem, and used your solution.

I couldn't comment on the CR, so here are two suggestions:
1. Add a validator. This would be helpful for all users. How many times have we all written code locally that fails once we try it for real? [The syntax here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html) could at least be added to a very simple regex, and if the user inputs something forbidden, i.e., a `!`, they could be directed to this same documentation with a message like, unsupported char, etc...
2. For the comparison variable type, you could enforce the options like so: `comparison: '=' | '!='`
```JavaScript
public static regexValue(jsonField: string, comparison: '=' | '!=', value: string): JsonPattern {
  return new JSONRegexPattern(jsonField, comparison, value);
}
```

**Comment by amrawadk** [comment]
@jan-xyz just wanted to say thank you for taking the time to document your solution, it helped me as well.

**Comment by jan-xyz** [comment]
Nice! I am also planning to still do the remaining work for the integration test. It's just very busy at the moment and I didn't find the time to get to it again.

**Comment by prplecake** [comment]
@jan-xyz Thanks for the solution!

**Comment by jan-xyz** [comment]
@JevinAnderson I am not too sure on how to validate the value, I did add the suggested restrictions for the comparison though. That should catch quite a few errors. If you have a suggestion on how to further check the value then let me know.

**Comment by paulhcsun** [comment]
Hi @jan-xyz, apologies for that inconvenience, I've unlocked the conversation on the PR. Thanks for adding the integ test, I will pass this to another team member to take a look.
